docs/src/examples/components/Dropdown/Usage/DropdownExampleControlled.shorthand.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react'
+import { Dropdown, Flex, Text } from '@stardust-ui/react'
+
+const inputItems = ['Bruce Wayne', 'Natasha Romanoff', 'Steven Strange', 'Alfred Pennyworth']
+
+class DropdownExampleControlled extends React.Component {
+  state = { open: false }
+
+  render() {
+    const open = this.state.open
+    return (
+      <Flex gap="gap.large" vAlign="center">
+        <Dropdown
+          open={open}
+          onOpenChange={this.handleOpenChange}
+          items={inputItems}
+          placeholder="Select your hero"
+        />
+        <Text weight="semibold" content={`Dropdown open state is: "${open}"`} />
+      </Flex>
+    )
+  }
+
+  handleOpenChange = (e, { open }) => {
+    this.setState({ open })
+  }
```

Comment:
Even in React docs event handlers are placed before `render()`, https://reactjs.org/docs/handling-events.html

Reply:
### I think there are 2 POVs here:

1. React with JS -> your example
2. TypeScript and OOP -> my approach

It is true that in your [reference](https://reactjs.org/docs/handling-events.html) the event handlers are placed before `render`, but we have to take into account that:
- your reference is JS code with ES6 (where classes DO NOT have access modifiers);
- we're writing TypeScript code (where classes DO have access modifiers which makes it more OOP-like). Here there is **a common practice to place public methods before protected and protected before private**. 

### Also, is that a good way to structure it necessarily (events handlers before `render`)?

One might argue that when you read the code you want to start with the entry points (the public methods, `render` in this case) and continue to scroll down as if it was a story, without too many jumps to the private methods that are called from the public ones (and constructor, that would also be placed at the beginning of the class).

I'm just expressing my thoughts here, **I'd like to add others to pitch in so we can stick with one convention. It's not what convention we choose that's important but having consistency.**

@mnajdova 
@kuzhelov 
@miroslavstastny 
@alinais  
@levithomason

Reply:
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-comp.md

My vote is to follow patterns that exists in React world to make the Stardust more friendly for community users.

Reply:
I did the order according to React convention for now, let's resolve for now and stick to what we have, tracking in #936